### PR TITLE
Fix GPU memory leak from missing .detach() in model wrappers

### PR DESCRIPTION
**Pull request by reillyosadchey**
## Summary

- Several model wrappers return tensors still attached to the autograd computation graph, causing the entire forward-pass graph to be retained in GPU memory across simulation steps
- Added `.detach()` calls to model outputs in `fairchem`, `orb`, `metatomic`, `fairchem_legacy`, and `graphpes_framework`
- The `fairchem_legacy` wrapper also used `.clone()` without `.detach()` on inputs, retaining graph references via `self.data_object`

## Details

| Model | Issue | Severity |
|---|---|---|
| `fairchem.py` | energy/forces/stress not detached | High |
| `orb.py` | predictions + conservative forces/stress not detached | High |
| `metatomic.py` | forces/stress not detached (energy was correct) | Medium |
| `fairchem_legacy.py` | `.clone()` without `.detach()` on inputs | Medium |
| `graphpes_framework.py` | external library predictions not detached | Low |

Without `.detach()`, `.to(dtype=...)` and `.view()` create new graph nodes that still reference the full forward pass. Integrators store these into `state.energy`/`state.forces`/`state.stress`, keeping the graph alive until the next step overwrites them — and longer if trajectory loggers or MC routines `.clone()` the state.

## Test plan

- [ ] Existing tests pass (no functional change — detach only affects graph retention)
- [ ] Verify GPU memory stays flat over long simulations with FairChem/Orb models

### CODE REVIEW

**Comment by CompRhys** [comment]
Thank you for the PR I will merge it once tests and lint pass.

It is worth noting that we are attempting to move all the models you fixed this for to external posture. These fixes may also need to be applied upstream.

In the pair potential backend refactor I am doing I also included a [retrain graph argument ](https://github.com/TorchSim/torch-sim/pull/488/changes#diff-e6724f79e75d566c0accf7d9f7e1b2dc76aa08b94881a644f8a2659757a6ad76R418) to allow for diff-sim tutorial. I think that's a very niche use so wouldn't add the flag here but just an FYI

File: torch_sim/models/graphpes_framework.py [hunk 1]
```diff
@@ -185,4 +185,5 @@ def forward(self, state: ts.SimState, **_kwargs: object) -> dict[str, torch.Tens
         if not isinstance(cutoff, torch.Tensor):
             raise TypeError("GraphPES model cutoff must be a tensor")
         atomic_graph = state_to_atomic_graph(state, cutoff)
-        return self._gp_model.predict(atomic_graph, self._properties)  # type: ignore[return-value]
+        preds = self._gp_model.predict(atomic_graph, self._properties)  # ty: ignore[call-non-callable]
+        return {k: v.detach() for k, v in preds.items()}
```
Comment (CompRhys) [inline torch_sim/models/graphpes_framework.py:189]: It feels cleaner to be to detatch everything like this all in one line at the end? Could you update all the models to follow this pattern?

Comment (CompRhys) [reply torch_sim/models/graphpes_framework.py:189]: if it's already detached then the op is idempotent and so no harm.

Comment (reillyosadchey) [reply torch_sim/models/graphpes_framework.py:189]: Good call

**Comment by CompRhys** [comment]
Thanks!
